src/api/Shrooms.Domain/Services/FilterPresets/FilterPresetService.cs

```diff
@@ -0,0 +1,259 @@
+        {
+            await _filterPresetUpdateCreateLock.WaitAsync();
+
+            try
+            {
+                _validator.CheckIfMoreThanOneDefaultPresetExists(updateDto);
+                _validator.CheckIfFilterPresetsContainsUniqueNames(updateDto.PresetsToUpdate);
+                _validator.CheckIfFilterPresetsContainsUniqueNames(updateDto.PresetsToCreate);
+                
+                await UpdatePresetsAsync(updateDto);
+                
+                var deletedPresets = await DeleteAsync(updateDto);
+
+                // Need to call this after deletion, because we receive preset Ids that need to be deleted and not presets themselves
+                await _validator.CheckIfUpdatedAndAddedPresetsHaveUniqueNamesExcludingDeletedPresetsAsync(updateDto, deletedPresets);
+
+                var createdPresets = CreatePresets(updateDto);
+
+                if (NewDefaultPresetWasSet(createdPresets, updateDto.PresetsToUpdate))
+                {
+                    await ChangeCurrentDefaultFilterToNonDefaultAsync(updateDto.PageType, updateDto.UserOrg.OrganizationId);
+                }
+
+                await _uow.SaveChangesAsync(updateDto.UserOrg.UserId);
+                
+                return new UpdatedFilterPresetDto
+                {
+                    // Ids are assigned after SaveChangesAsync()
+                    CreatedPresets = createdPresets.Select(preset => new FilterPresetDto
+                    {
+                        Id = preset.Id,
+                        Name = preset.Name,
+                        IsDefault = preset.IsDefault,
+                        Filters = JsonConvert.DeserializeObject<IEnumerable<FilterPresetItemDto>>(preset.Preset)
+                    }),
+                    UpdatedPresets = updateDto.PresetsToUpdate,
+                    DeletedPresets = deletedPresets
+                };
+            }
+            finally
+            {
+                _filterPresetUpdateCreateLock.Release();
+            }
+        }
+
+        public async Task<IEnumerable<FiltersDto>> GetFiltersAsync(FilterType[] filterTypes, int organizationId)
+        {
+            _validator.CheckIfFilterTypesContainsDuplicates(filterTypes);
+            _validator.CheckIfFilterTypesAreValid(filterTypes);
+
+            var filtersDtos = new List<FiltersDto>();
+
+            foreach (var type in filterTypes)
+            {
+                var filters = await GetFiltersQueryByFilterType(type).ToListAsync();
+                var filtersDto = new FiltersDto
+                {
+                    FilterType = type,
+                    Filters = filters,
+                };
+
+                filtersDtos.Add(filtersDto);
+            }
+
+            return filtersDtos;
+        }
+
+        private bool NewDefaultPresetWasSet(IEnumerable<FilterPreset> createdPresets, IEnumerable<FilterPresetDto> updatedPresets)
```

Comment:
Sounds more natural:
```suggestion
        private bool IsNewDefaultPresetSet(IEnumerable<FilterPreset> createdPresets, IEnumerable<FilterPresetDto> updatedPresets)
```